If badmailfrom contains ".example.com" then reject mail from
"anything@any.thing.example.com"

http://tomclegg.net/qmail/qmail-badmailfrom-wildcard.patch

--- qmail-smtpd.c~	Thu Apr 17 16:08:41 2003
+++ qmail-smtpd.c	Thu Apr 17 20:04:16 2003
@@ -215,8 +215,13 @@
   if (!bmfok) return 0;
   if (constmap(&mapbmf,addr.s,addr.len - 1)) return 1;
   j = byte_rchr(addr.s,addr.len,'@');
-  if (j < addr.len)
+  if (j < addr.len) {
     if (constmap(&mapbmf,addr.s + j,addr.len - j - 1)) return 1;
+    for (j++; j < addr.len; j++)
+      if (addr.s[j] == '.') {
+	if(constmap(&mapbmf,addr.s + j,addr.len - j - 1)) return 1;
+      }
+  }
   return 0;
 }
 
--- /tmp/qmail-1.03/qmail-smtpd.8	Mon Jun 15 03:53:16 1998
+++ qmail-smtpd.8	Sun Aug  1 15:19:19 2004
@@ -49,6 +49,10 @@
 .BR @\fIhost ,
 meaning every address at
 .IR host .
+A line of the form
+.BR .\fIdomain
+means every address at every host ending with
+.IR .domain .
 .TP 5
 .I databytes
 Maximum number of bytes allowed in a message,
